provider/hybrik/hybrik.go

```diff
@@ -0,0 +1,500 @@
+			hwrapper.Connection{
+				From: manifestFromConnections,
+				To: hwrapper.ConnectionTo{
+					Success: []hwrapper.ToSuccess{
+						{Element: "manifest_creator"},
+					},
+				},
+			},
+		)
+
+	}
+
+	resp, err := json.Marshal(cj)
+	if err != nil {
+		return "", err
+	}
+
+	return string(resp), nil
+}
+
+func (hp *hybrikProvider) JobStatus(job *db.Job) (*provider.JobStatus, error) {
+
+	ji, err := hp.c.GetJobInfo(job.ProviderJobID)
+	if err != nil {
+		return &provider.JobStatus{}, err
+	}
+
+	var status provider.Status
+	switch ji.Status {
+	case active:
+		fallthrough
+	case activeRunning:
+		fallthrough
+	case activeWaiting:
+		status = provider.StatusStarted
+	case queued:
+		status = provider.StatusQueued
+	case completed:
+		status = provider.StatusFinished
+	case failed:
+		status = provider.StatusFailed
+	}
+
+	return &provider.JobStatus{
+		ProviderJobID: job.ProviderJobID,
+		ProviderName:  hp.String(),
+		Progress:      float64(ji.Progress),
+		Status:        status,
+	}, nil
+}
+
+func (hp *hybrikProvider) CancelJob(id string) error {
+	return hp.c.StopJob(id)
+}
+
+func (hp *hybrikProvider) CreatePreset(preset db.Preset) (string, error) {
+	var minGOPFrames, maxGOPFrames, gopSize int
+
+	gopSize, err := strconv.Atoi(preset.Video.GopSize)
+	if err != nil {
+		return "", err
+	}
+
+	if preset.Video.GopMode == "fixed" {
+		minGOPFrames = gopSize
+		maxGOPFrames = gopSize
+	} else {
+		maxGOPFrames = gopSize
+	}
+
+	container := ""
+	for _, c := range hp.Capabilities().OutputFormats {
+		if preset.Container == c || (preset.Container == "m3u8" && c == hls) {
+			container = c
+		}
+	}
+
+	if container == "" {
+		return "", ErrUnsupportedContainer
+	}
+
+	bitrate, err := strconv.Atoi(preset.Video.Bitrate)
+	if err != nil {
+		return "", ErrBitrateNan
+	}
+
+	audioBitrate, err := strconv.Atoi(preset.Audio.Bitrate)
+	if err != nil {
+		return "", ErrBitrateNan
+	}
+
+	var videoWidth *int
+	var videoHeight *int
+
+	if preset.Video.Width != "" {
+		var presetWidth int
+		presetWidth, err = strconv.Atoi(preset.Video.Width)
+		if err != nil {
+			return "", ErrVideoWidthNan
+		}
+		videoWidth = &presetWidth
+	} else {
+		videoWidth = nil
+	}
+
+	if preset.Video.Height != "" {
+		var presetHeight int
+		presetHeight, err = strconv.Atoi(preset.Video.Height)
+		if err != nil {
+			return "", ErrVideoHeightNan
+		}
+		videoHeight = &presetHeight
+	} else {
+		videoHeight = nil
+	}
+
+	videoProfile := strings.ToLower(preset.Video.Profile)
+	videoLevel := preset.Video.ProfileLevel
+
+	// TODO: Understand video-transcoding-api profile + level settings in relation to vp8
+	// For now, we will omit and leave to encoder defaults
+	if preset.Video.Codec == "vp8" {
+		videoProfile = ""
+		videoLevel = ""
+	}
+
+	p := hwrapper.Preset{
+		Key:         preset.Name,
+		Name:        preset.Name,
+		Description: preset.Description,
+		Kind:        "transcode",
+		Path:        "video-transcoding-api-presets",
```

Comment:
Presets can be organized into folders, and the path is essentially the virtual folder path for the preset. It's purely for organizational purposes. In our web interface, you'll be able to see these presets and they'll be in a folder called "video-transcoding-api-presets". You can also query for presets by path via our API.

We'll expose this via an optional env variable.